### PR TITLE
Second PR with implementations of new views and user authentication

### PR DESCRIPTION
**Pull request by TheLastMercy**
Regex just doesn't work. If anyone got any ideas, you are more then welcome to try them. Everything I tried just destroyed whole 
structure...

### CODE REVIEW

File: frontend/feedbacks/views.py [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+from .models import *
 import requests
 
 # Create your views here.
 def eval_feedback_view(request, *args, **kwargs):
     feedback = request.GET["feedback"]
-    eval_id = request.GET["eval_id"]
+    eval_block_id = request.GET["eval_block_id"]
 
     Evaluation_Feedback.objects.create(feedback=feedback, evaluation=eval_id)
```
Comment (Dzeniks) [inline frontend/feedbacks/views.py:11]: I dont know, where you getting eval_id, so i assume its should be eval_block_id
```suggestion
    Evaluation_Feedback.objects.create(feedback=feedback, evaluation=eval_block_id)
```

File: frontend/userExtensions/views.py [hunk 2]
```diff
@@ -71,10 +71,18 @@ def login_view(request, *args, **kwargs):
     body_unicode = request.body.decode('utf-8')
     data = json.loads(body_unicode)
    
-    username = data.get("username_mail")
+    username_email = data.get("username_email")
     password = data.get("password")
 
-    user = auth.authenticate(username=username, password=password)
+    if "@" in username_email:
```
Comment (Dzeniks) [inline frontend/userExtensions/views.py:77]: If you want to check if it's mail, the @ sign should be disabled in the name, just a reminder.

File: svelteSource/src/lib/validationOutput/ClaimBlock.svelte [hunk 3]
```diff
@@ -44,7 +44,7 @@
       disabled = true;
 
       const csrftoken = getCookie("csrftoken");
-      let url = `/rag_evaluation?text=${text}`;
+      let url = `/rag/eval_BM25?text=${text}`;
```
Comment (Dzeniks) [inline svelteSource/src/lib/validationOutput/ClaimBlock.svelte:47]: Wrong url, and if its positible you should use parameter as argument in request

base_url = 'https://example.com/api'
params = {
    'param1': 'value1',
    'param2': 'value2',
}
response = requests.get(base_url, params=params)

```suggestion
      let url = `/rag/eval_bm25?text=${text}`;
```

File: frontend/feedbacks/views.py [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
 
+    return HttpResponse(status=402)
+
+def user_feedback_view(request, *args, **kwargs):
+    feedback = request.GET["feedback"]
+    user = request.user
+
+    User_Feedback.objects.create(feedback=feedback, user=user)
 
     return HttpResponse(status=402)
```
Comment (Dzeniks) [inline frontend/feedbacks/views.py:21]: This should always return 402 code ? xD

Comment (TheLastMercy) [reply frontend/feedbacks/views.py:21]: for now, yes xD

File: svelteSource/src/lib/validationOutput/ClaimBlock.svelte (hunk 3 above)
Comment (TheLastMercy) [reply svelteSource/src/lib/validationOutput/ClaimBlock.svelte:47]: this is Stefik's issue, I only rewrited his suggestion into url :D

File: frontend/userExtensions/views.py (hunk 2 above)
Comment (TheLastMercy) [reply frontend/userExtensions/views.py:77]: I'll fix it, thx.

File: frontend/feedbacks/views.py (hunk 1 above)
Comment (TheLastMercy) [reply frontend/feedbacks/views.py:11]: You are correct, I forgot to change that. Thanks a lot.

**Review by Dzeniks — APPROVED** [review]

File: frontend/evaluations/views.py [hunk 5]
```diff
@@ -3,114 +3,116 @@
...
+    # * Check for authentication
+    user = request.user
+    if user.is_authenticated:
+        validated_text = requests.get(backend_url, params={"text" : input_text})
+
+        if validated_text.status_code == 503:
+            request_queue.put((request, input_text, backend_url))
+            return JsonResponse({"status" : 503})
+        
+        while not request_queue.empty():
```
Comment (Dzeniks) [inline frontend/evaluations/views.py:46]: I realized that it is not a bad thing to make an interval that waits before asking. So you don't keep calling my endpoint.

File: frontend/evaluations/views.py [hunk 6]
```diff
@@ -3,114 +3,116 @@
...
-            nei=evaluation.get("nei"),
-            evidence=evaluation.get("evidence")
-        )
-        evaluation["id"] = new_evaluation.id # ! Adding id to the obtained JSON -> passing to feedbacks app
-        evaluation["evaluation_block"] = new_evaluation_block.id
 
-    return JsonResponse({"validated" : validated_text})
+        # * New Evaluation Block creation
+        whole_claim = " ".join([claim["claim"] for claim in validated_text.json()])
+        new_evaluation_block = Evaluation_block.objects.create(user=user, claims=whole_claim)
```
Comment (Dzeniks) [inline frontend/evaluations/views.py:66]: Add some error handling

```suggestion
try:
        whole_claim = " ".join([claim["claim"] for claim in validated_text.json()])
        new_evaluation_block = Evaluation_block.objects.create(user=user, claims=whole_claim)
except:
        print("Error")
```

**Review by OriginalStefikO — CHANGES_REQUESTED** [review]
Seems good to me now, just @Dzeniks comments need to be acknowledged

**Pull request merged**